app/controllers/pages_controller.rb

```diff
@@ -2,13 +2,14 @@
 
 class PagesController < ApplicationController
   def home
-    @products = Product.where(query_params[:filter])
     @categories_select_params = Category.select_params
+    @brands_select_params = Brand.select_params
+    @products = Product.filtered(filter_params)
```

Comment:
Muszę się pochwalić, bo udało mi się chyba ładnie ogarnąć takiego potwora 🥳 
```
def home
    @products = Product.all
    if filter_params[:filter]
      @products = @products.where(category_id: filter_params[:filter][:category_id]) if filter_params[:filter][:category_id].present?
      @products = @products.where(brand_id: filter_params[:filter][:brand_id]) if filter_params[:filter][:brand_id].present?

    end
    @categories = Category.all
    @brands = Brand.all
  end
```

Reply:
ladnie :). Jest lepszy sposob aby to zrobic: google `searchlight`